storage/database/db_manager.go

```diff
@@ -315,20 +337,48 @@ func singleDatabaseDBManager(dbc *DBConfig) (DBManager, error) {
 	return dbm, nil
 }
 
+// getMiscDB returns misc DBManager
+func getMiscDB(dbc *DBConfig) Database {
```

Comment:
How about changing this function name to `newMiscDB`.
It always creates a new database.